docs/API/fliplet-oauth2.md

```diff
@@ -0,0 +1,149 @@
+# `Fliplet.OAuth2` (Beta)
+
+The `fliplet-oauth2` package contains helpers for standardizing requests to OAuth2 web services with a client-side integration.
+
+**Note: The Fliplet OAuth2 library is Currently in beta and only works in native apps. We suggest specifying the library version using `fliplet-oauth2:0.1` when including the Fliplet OAuth2 library to avoid the functionality breaking as the feature undergoes further development.**
+
+To configure an OAuth2 provider, call `Fliplet.OAuth2.configure()` with the provider specificaions and use `Fliplet.OAuth2(provider)` with the specified provider name to access the available methods, e.g. `Fliplet.OAuth2(provider).api(path)`.
+
+```js
+// Configure providers
+Fliplet.OAuth2.configure(provider, configuration);
+// Make API requests using configured provider
+Fliplet.OAuth2(provider).api(path)
+  .then(function (response) {
+    // Process response from API provider as necessary
+  });
+```
+
+## Examples
+
+### 1. Configure a connection with the GitHub provider
+
+*Note: The provider configuration is likely executed via the Global JS code in Fliplet Studio.*
+
+```js
+Flipet.OAuth2.configure('github', {
+  authUrl: 'http://github.com/login/oauth/authorize', // from OAuth2 provider
+  grantType: 'implicit', // as supported by OAuth2 provider
+  grantUrl: 'https://github.com/login/oauth/access_token', // from OAuth2 provider
+  baseUrl: 'https://api.github.com/', // from OAuth2 provider
+  clientId: '4c867cd2a96541883322', // from OAuth2 provider
+  clientSecret: '5f66b7a07842570936512097ec255721b259b16a', // from OAuth2 provider
```

Comment:
this is not our secret key, right? if so, we need to invalidate it since this repo is public :)